### PR TITLE
Use ROS-independent version of ICI

### PR DESCRIPTION
**Pull request by marip8**
Changes per commit message to address #834

### CODE REVIEW

**Comment by marip8** [comment]
It looks like the Focal builds are failing on the octomap unit test due to the issue documented in #828. Also the Bionic build can't find `gtest` (maybe because it is no longer being provided by `ros_industrial_cmake_boilerplate`?).

**Comment by Levi-Armstrong** [comment]
It does not look like the libraries are provided on 18.04. I am fine with removing this CI build because it will be EOL in April.

File: dependencies.repos [hunk 1]
```diff
@@ -6,3 +6,11 @@
     local-name: opw_kinematics
     uri: https://github.com/Jmeyer1292/opw_kinematics.git
     version: 0.5.0
+- git:
+    local-name: fcl
+    uri: https://github.com/flexible-collision-library/fcl.git
+    version: 0.6.1
```
Comment (Levi-Armstrong) [inline dependencies.repos:12]: Why did you have to add fcl?

Comment (marip8) [reply dependencies.repos:12]: Version 0.6.1 is not distributed and `tesseract_collision` requires at least that version

Comment (Levi-Armstrong) [reply dependencies.repos:12]: Can we not install it from ros?

Comment (marip8) [reply dependencies.repos:12]: My hope is to make this a fully ROS-independent build to support either ROS1 or ROS2 downstream repos, so no...

Comment (Levi-Armstrong) [reply dependencies.repos:12]: Ah, makes sense.

**Comment by marip8** [comment]
I merged a few of the jobs together to reduce code duplication. It's set up now to run all Ubuntu and Windows jobs in a matrix to support the same build steps for multiple distributions. I also merged the two code quality jobs (code coverage and clang tidy) in a matrix. It was getting difficult to make sure all the variables were set correctly, so this should help make introducing jobs for new distros easier

**Comment by Levi-Armstrong** [comment]
Looks good!

**Comment by marip8** [comment]
Looks like something is wrong with the Windows build since it's failed the last couple of merged PRs, but this PR should be ready for review

**Comment by Levi-Armstrong** [comment]
Windows is failing because I updated the ros_industrial_cmake_bolilerplate version and it no longer can find gtest.

**Comment by marip8** [comment]
> I would just add gtest to the vcpkg list

That apparently doesn't work, unless it's under a different name than `ros-industrial-cmake-boilerplate`

**Comment by Levi-Armstrong** [comment]
> > I would just add gtest to the vcpkg list
> 
> That apparently doesn't work, unless it's under a different name than `ros-industrial-cmake-boilerplate`. I also can't find it on the [`vcpkg` website](https://vcpkg.io/en/packages.html)

Need to add `gtest` to the list.

**Comment by Levi-Armstrong** [comment]
https://github.com/microsoft/vcpkg/tree/master/ports/gtest

**Comment by marip8** [comment]
Ah my mistake; was stuck on RICB for some reason

**Comment by Levi-Armstrong** [comment]
The issue now is that octomap repository is to being build first. I not sure why but it could be because it has a top level CMakeLists.txt file and colcon is not searching deeper to the sub directories which have a package.xml.

**Comment by Levi-Armstrong** [comment]
I would just add a separate rosinstall file for windows and remove fcl and octomap. Then add fcl and octomap back to vcpkg list.

**Comment by Levi-Armstrong** [comment]
LGTM, I recommend squash merging this when you are ready

**Pull request merged**